Realm/RLMRealm.mm

```diff
@@ -196,16 +187,26 @@ static void clearRealmCache() {
         for (NSMapTable *map in s_realmsPerPath.allValues) {
             [map removeAllObjects];
         }
-        s_realmsPerPath = [NSMutableDictionary dictionary];
+        [s_realmsPerPath removeAllObjects];
     }
 }
 
+static NSMutableDictionary *s_listenersPerPath = [NSMutableDictionary new];
+// must be called with s_listenersPerPath locked
+static RLMChangeListener *listenerForPath(NSString *path, BOOL inMemory, BOOL create=YES) {
```

Comment:
Should split this into two methods `startListening(Realm *realm)` and `stopListening(Realm *realm)`